### ISSUE TITLE
Style/ConditionalAssignment: use the return of the conditional for variable assignment and comparison incorrectly warning on case statements

### ISSUE THREAD

**Issue by TomasBarry** [rubocop-hq/rubocop#6245]
## Expected behavior

Case statement with an else clause should not throw the `Style/ConditionalAssignment: use the return of the conditional for variable assignment and comparison` error

## Actual behavior

Case statement with an else clause throws the `Style/ConditionalAssignment: use the return of the conditional for variable assignment and comparison` error

## Steps to reproduce the problem

Example code to recreate error:

```ruby
some_var = 0
case some_var
when 0 then some_var <= 1
when 1 then some_var <= 2
when 2 then some_var <= 3
when 3 then some_var <= 4
when 4 then some_var <= 5
else some_var <= 6
end
```

This error can be solved by adding `&& true` to any of the conditions, but this shouldn't be required

## RuboCop version

```
$ [bundle exec] rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.3.1 x86_64-darwin17)
```

**Comment by Drenmi** [comment]
Thanks for the report. 🙇 Problem verified on `master` and labelled as a bug.

**Comment by TomasBarry** [comment]
Hey @Drenmi, is there a plan/bug/timeline to fix this bug?

**Comment by Drenmi** [comment]
Here's a minimal test case for whomever wants to tackle this:

```ruby
context 'with a non-assigned case statement with an else clause' do
  let(:config) do
    RuboCop::Config.new('Style/ConditionalAssignment' => {
                          'Enabled' => true,
                          'EnforcedStyle' => 'assign_to_condition',
                          'SupportedStyles' => %w[assign_to_condition
                                                  assign_inside_condition]
                        },
                        'Metrics/LineLength' => {
                          'Max' => 80,
                          'Enabled' => true
                        })
  end

    it 'does not register an offense' do
      expect_no_offenses(<<-RUBY.strip_indent)
        case foo
        when 0 then foo <= 1
        else foo <= 2
        end
      RUBY
    end
  end
```

**Comment by TomasBarry** [comment]
@Drenmi, thanks for getting back. I'll take a look!

**Comment by TomasBarry** [comment]
This is still an issue in Rubocop

**Comment by TomasBarry** [comment]
This is still an issue in Rubocop

**Comment by buehmann** [comment]
Hi @TomasBarry, I have trouble understanding what the bug is supposed to be here. The cop seems to behave as designed. Could you explain what the issue is for you?

**Comment by shubham-420** [comment]
Still issue in Rubocop

**Comment by jonas054** [comment]
> Case statement with an else clause should not throw the Style/ConditionalAssignment: use the return of the conditional for variable assignment and comparison error

Why not? It's only when you have an `else` branch that you can safely change

```ruby
some_var = 0
case some_var
when 0 then some_var <= 1
when 1 then some_var <= 2
when 2 then some_var <= 3
when 3 then some_var <= 4
when 4 then some_var <= 5
else some_var <= 6
end
```

into

```ruby
some_var = 0
some_var <= case some_var
            when 0 then 1
            when 1 then 2
            when 2 then 3
            when 3 then 4
            when 4 then 5
            else 6
            end
```

**Comment by jonas054** [comment]
Neither @buehmann or myself can understand what's wrong with the current behavior, and since no-one has answered our questions I'm closing the issue. Comment with an explanation if you disagree.
